go/arrow/ipc/writer.go

```diff
@@ -562,6 +562,36 @@ func (w *recordEncoder) visit(p *Payload, arr arrow.Array) error {
 		p.body = append(p.body, voffsets)
 		p.body = append(p.body, values)
 
+	case *arrow.LargeBinaryType:
```

Comment:
Isn't this exactly the same code as for BinaryType above? Can this perhaps be mutualized?

Reply:
Incidentally, same question for String/LargeString and List/LargeList